dashboard/src/components/api/environment/kubernetes-machine-recipe-parser.ts

```diff
@@ -150,12 +150,6 @@ export class KubernetesMachineRecipeParser implements IParser {
   }
 
   validate(recipe: ISupportedListItem ): void {
-    if (!recipe || !recipe.kind) {
-      throw new TypeError(`Recipe item should contain a 'kind' section.`);
-    }
-    if (!recipe.apiVersion) {
-      throw new TypeError(`Recipe item should contain 'apiVersion' section`);
-    }
```

Comment:
I'm not sure this is accurate -- the kubernetes machine recipe should be valid kubernetes; without `kind` and `apiVersion` fields it is not. 

Regardless, we can weaken validation if necessary.

Reply:
To be clear, I want to understand the root cause of the issue being solved before weakening validation.

Reply:
Machine recipe - client-side abstraction. It is a part of the valid kubernetes recipe (just a container). And we cower them with some information from the parent pod before showing to end user. 

In any case, it is just a part of the valid kubernetes recipe. And we should check this part only.